File: src/booking/schemas/event.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class ListQuery(BaseModel):
    limit: int = Field(default=20, ge=1, le=100)
    offset: int = Field(default=0, ge=0)


class EventShort(BaseModel):
    id: uuid.UUID
    title: str
    starts_at: datetime
    venue: str | None
    age_rating: str | None
    banner_small_url: str | None
    free_tickets: int | None


class EventListResponse(BaseModel):
    items: list[EventShort]
    total: int


class EventDetail(BaseModel):
    id: uuid.UUID
    title: str
    description: str | None
    starts_at: datetime
    duration_min: int | None
    age_rating: str | None
    venue: str | None
    price: Decimal | None
    status: str
    banner_small_url: str | None
    banner_large_url: str | None
    show_free_tickets: bool
    sale_paused: bool
    free_tickets: int | None


class PageResponse(BaseModel):
    slug: str
    title: str
    content: str

File: frontend/src/types/event.ts
export type EventStatus = 'draft' | 'on_sale' | 'paused' | 'cancelled' | 'completed' | 'moved';

export interface EventRead {
  id: string;
  title: string;
  description: string | null;
  starts_at: string;
  duration_min: number | null;
  age_rating: string | null;
  venue: string | null;
  price: string | null;
  status: EventStatus;
  banner_small_url: string | null;
  banner_large_url: string | null;
  show_free_tickets: boolean;
  sale_paused: boolean;
  cloned_from_id: string | null;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface TicketTypeSeed {
  name: string;
  price: number;
  quota: number;
}

export interface EventCreate {
  title: string;
  description?: string | null;
  starts_at: string;
  duration_min?: number | null;
  age_rating?: string | null;
  venue?: string | null;
  banner_small_url?: string | null;
  banner_large_url?: string | null;
  show_free_tickets?: boolean;
  sale_paused?: boolean;
  ticket_types?: TicketTypeSeed[] | null;
}

export type EventUpdate = Partial<EventCreate>;

export interface EventMove {
  starts_at: string;
}

export interface TicketTypeRead {
  id: string;
  event_id: string;
  name: string;
  price: string;
  quota: number;
  sold: number;
  deleted_at: string | null;
}

export interface TicketTypeCreate {
  name: string;
  price: number;
  quota: number;
}

export type TicketTypeUpdate = Partial<TicketTypeCreate>;
